sdk/resourcemanager/Azure.ResourceManager/src/ApiVersionsBase.cs

```diff
@@ -137,10 +137,22 @@ public int CompareTo(string other)
                     return 1;
                 }
 
-                return string.Compare(thisPreviewPart, otherPreviewPart, StringComparison.InvariantCulture);
+                // string.Compare doesn't ensure the return value is 0, 1, -1, so here we need to normalize it
```

Comment:
will these code be replaced by the auto-generated code @fengzhou-msft ?

Reply:
Only code under `PseudoGenerated` will be auto-generated and this file will not be generated.